packages/viz/src/components/LineChart/LineChart.tsx

```diff
@@ -459,43 +459,44 @@ export const HvLineChart = ({
     measures,
   ]);
 
-  const renderTooltip = (
-    params: {
-      seriesName: string;
-      value: (string | number)[];
-      encode: { [key: string]: number[] };
-      color: string;
-      dimensionNames: string[];
-    }[],
-    single: boolean,
-    msr: MeasuresField | MeasuresField[],
-    cls?: typeof classes,
-    formatter?: (value?: string | number) => string
-  ) => {
-    const tooltipTitle = params[0].value[params[0].encode.x[0]];
-
-    if (single) {
-      const measure = getMeasure(
-        params[0].dimensionNames[params[0].encode.y[0]],
-        msr
-      );
-      const value = params[0].value[params[0].encode.y[0]];
-      const formattedValue =
-        typeof measure !== "string" && measure.valueFormatter
-          ? measure.valueFormatter(value)
-          : formatter
-          ? formatter(value)
-          : value;
-
-      return `
+  const renderTooltip = useCallback(
```

Comment:
This function (along with `getMeasure`) don't depend on component state outside the params - they're pure functions!.
We should be moving them outside `LineChart`, not memoizing it

Reply:
This was already done in the other PR. Please rebase